fish-rust/src/common.rs

```diff
@@ -2056,17 +2065,81 @@ pub fn fputws(s: &wstr, fd: RawFd) {
     wwrite_to_fd(s, fd);
 }
 
+#[cfg(test)]
 mod tests {
-    use crate::common::{
-        escape_string, str2wcstring, wcs2string, EscapeStringStyle, ENCODE_DIRECT_BASE,
-        ENCODE_DIRECT_END,
-    };
-    use crate::wchar::widestrs;
-    use crate::wutil::encoding::{wcrtomb, zero_mbstate, AT_LEAST_MB_LEN_MAX};
+    use super::*;
+
+    use crate::future_feature_flags::future_feature_flags_init;
+    use crate::tests::make_locale_sensitive;
     use rand::random;
 
+    /// The number of tests to run.
+    const ESCAPE_TEST_COUNT: usize = 100_000;
```

Comment:
```suggestion
    const ESCAPE_TEST_COUNT: usize = 10_000;
```
With the current speed it's not worth doing 100k tests at `opt-level=0`.

Reply:
There is an argument for splitting the str2wcstring, (un)escape and friends in a separate crate and then only running tests when code in that crate changes. We should probably use https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html to make that easier, but I don't wanna mess with the build system right now. We could then also compile at a higher opt-level for potentially faster tests?

Reply:
>With the current speed it's not worth doing 100k tests at opt-level=0.

That's fine.

> There is an argument for splitting the str2wcstring, (un)escape and friends in a separate crate and then only running tests when code in that crate changes.

Separate crates are *definitely* a thing for after the initial port.